apps/ntag/arduinoLibs/WString.cpp

```diff
@@ -0,0 +1,763 @@
+/*
```

Comment:
It would be easier to review this if you committed the original version before porting to posix.  I would then be able to see the differences rather than having to evaluate every line.  As a result, I just skimmed the file.

Reply:
The existing Arduino project this is from has too many files and really we only need what is necessary for compiling the stack.